### DISCUSSION TITLE
LAB 2 ap_start 沒有連結

### DISCUSSION THREAD

**Discussion by MODKWODK** [bol-edu/HLS-SOC-Discussions#71] [Q&A] [answered]
之前同學有在slack發問過的問題 
助教已經回答 但是還是看不懂解說
詢問 這邊的Port-level protocal 的AXI-Lite 該如何使用?
![Uploading image.png…]()

![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145253191/3ecdab91-0f4d-43f8-baae-03e12698bebd)

**Comment by hbx1241** [comment]
同學好，我想這段的意思是把 port=return 的 interface 改成 AXI-lite 而不是 ap_ctrl_none

**Reply by MODKWODK** [reply]
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145253191/771dddaa-ef51-494e-9230-3c215a3ab940)
ni
同學你好!
非常感謝你的回答
但我還不太理解 請問是這個部分嗎?

因為我這樣做了之後 出來的sample 還是有ap_start
如下圖
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145253191/125b3d94-f1c2-4546-a08f-4a39b9eaf5bf)

我推估是不是這個地方做錯了
下圖藍色的部分是否需要更改?
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145253191/df3c405b-319d-473e-b9c6-470a6652ba46)

**Reply by Lucashien** [reply]
我也有這個問題，已將port=return 的 interface 改成 AXI-Lite，但仍然會有此問題QQ
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/64431591/99b47eac-dee9-414c-b54e-79d630ffdb84)

**Reply by Lucashien** [reply]
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/64431591/d660569b-92d3-426c-9772-ba1618b34324)
我直接留白，ap_start就不見了，給你參考

但我不知為啥我的simulation會一直呈現high-z
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/64431591/71a9ebf9-dece-4458-bac6-15577b624536)

**Reply by MODKWODK** [reply]
同學你好~~!!

我後來也是直接留白 就可以完成了
除了 m_axis 有另外設定外，其餘的s_axis都不另外做設定

**Comment by tim21525** [comment, marked as answer]
![348AD6D1-0837-4AC0-B8EE-9625571A2462](https://github.com/bol-edu/HLS-SOC-Discussions/assets/145553529/c27c5955-c319-46ee-b883-41fbd6c4aa31)

我是把fir_n11_maxi的depth留白，好像就能解決這個問題

**Comment by Josephood7** [comment]
![image](https://github.com/bol-edu/HLS-SOC-Discussions/assets/144367420/faa75a0d-efc5-4ed8-a2a5-7fe9b6e6908b)
我的理解是Vitis這個軟體在建立AXI4-Lite的時候，會自動把ap_ctrl的所有control protocol捆在Interface上。
我的作法是把原本directive.tcl檔的程式碼用##comment掉或刪除變空白。用lab1的方式直接在.cpp檔裡面宣告INTERFACE
雖然沒有做到老師希望的用不同方式定義INTERFACE練習的用意，但最後成功把ap_ctrl刪除，且成功generate bit stream

#pragma HLS INTERFACE s_axilite port=regXferLeng
#pragma HLS INTERFACE s_axilite port=an32Coef
#pragma HLS INTERFACE m_axi port=pn32HPOutput depth=600 offset=slave
#pragma HLS INTERFACE m_axi port=pn32HPInput depth=600 offset=slave
#pragma HLS INTERFACE s_axilite port=return
